devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java

```diff
@@ -0,0 +1,608 @@
+     * <ul>
+     * <li>{@code deployment-pom.xml}</li>
+     * <li>{@code parent-pom.xml}</li>
+     * <li>{@code runtime-pom.xml}</li>
+     * <li>{@code Processor.java}</li>
+     * </ul>
+     * Note that you do not need to provide all of them. Files not available in your custom {@link #templatesUriBase}
+     * will be looked up in the default URI base {@value #DEFAULT_TEMPLATES_URI_BASE}. The default templates are
+     * maintained <a href=
+     * "https://github.com/quarkusio/quarkus/tree/master/devtools/maven/src/main/resources/create-extension-templates">here</a>.
+     */
+    @Parameter(defaultValue = DEFAULT_TEMPLATES_URI_BASE, required = true, property = "quarkus.templatesUriBase")
+    String templatesUriBase;
+
+    /** Encoding to read and write files in the current source tree */
+    @Parameter(defaultValue = DEFAULT_ENCODING, required = true, property = "quarkus.encoding")
+    String encoding;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        if (this.basedir == null) {
+            this.basedir = Paths.get(".").toAbsolutePath().normalize();
+        }
+        if (artifactId != null) {
+            artifactIdBase = artifactIdBase(artifactId);
+            artifactIdPrefix = artifactId.substring(0, artifactId.length() - artifactIdBase.length());
+            artifactId = BRACKETS_PATTERN.matcher(artifactId).replaceAll("");
+        } else if (artifactIdPrefix != null && artifactIdBase != null) {
+            artifactId = artifactIdPrefix + artifactIdBase;
+        } else {
+            throw new MojoFailureException(String.format(
+                    "Either artifactId or both artifactIdPrefix and artifactIdBase must be specified; found: artifactId=[%s], artifactIdPrefix=[%s], artifactIdBase[%s]",
+                    artifactId, artifactIdPrefix, artifactIdBase));
+        }
+
+        if (name != null) {
+            final int pos = name.lastIndexOf(nameSegmentDelimiter);
+            if (pos >= 0) {
+                nameBase = name.substring(pos + nameSegmentDelimiter.length());
+            } else {
+                nameBase = name;
+            }
+            namePrefix = name.substring(0, name.length() - nameBase.length());
+        } else {
+            if (nameBase == null) {
+                nameBase = toCapWords(artifactIdBase);
+            }
+            if (namePrefix == null) {
+                namePrefix = "";
+            }
+            if (nameBase != null && namePrefix != null) {
+                name = namePrefix + nameBase;
+            } else {
+                throw new MojoFailureException("Either name or both namePrefix and nameBase must be specified");
+            }
+        }
+
+        final Charset charset = Charset.forName(encoding);
+
+        final Path basePomXml = basedir.resolve("pom.xml");
+        if (Files.exists(basePomXml)) {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
```

Comment:
It's not a huge deal but for fyi we've been using `io.quarkus.maven.utilities.MojoUtils` to read models and related bits.

Reply:
There is no `MojoUtils.readPom()` override with an explicit encoding param. Thus -1.